trio/testing/_trio_test.py

```diff
@@ -25,6 +25,7 @@ def wrapper(**kwargs):
             clock = clocks[0]
         else:
             raise ValueError("too many clocks spoil the broth!")
-        return _core.run(partial(fn, **kwargs), clock=clock)
+        instruments = [i for i in kwargs.values() if isinstance(i, Instrument)]
```

Comment:
Shouldn't `instruments` be removed from `kwargs` as well?

```python
instruments = [kwargs.pop(k) for (k, v) in kwargs.items() if isinstance(v, Instrument)]
```

Otherwise `fn` will also receive the instruments still in `**kwargs`.

Reply:
This mechanism is typically for pytest fixtures, where you declare the fixture you want by having arguments in your function. Hence if you modify kwargs, the called function will no longer have the right number of arguments.

Reply:
Oh I see so you want to pass the instruments to `_core.run` *in addition* to `fn`, got it.

Sorry for the noise! 👍